common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -276,27 +276,55 @@ def _is_in_bulk_operation(self, course_key, ignore_case=False):
         return self._get_bulk_ops_record(course_key, ignore_case).active
 
 
-class ModuleStoreAssetInterface(object):
+class SortedAssetList(SortedListWithKey):
     """
-    The methods for accessing assets and their metadata
+    List of assets that is sorted based on an asset attribute.
     """
-    @contract(asset_list='SortedListWithKey', asset_id='AssetKey')
-    def _find_asset_in_list(self, asset_list, asset_id):
+    def __init__(self, filename_sort=True, **kwargs):
+        self.filename_sort = filename_sort
+        if filename_sort:
+            kwargs['key'] = itemgetter('filename')
+        super(SortedAssetList, self).__init__(**kwargs)
+
+    @contract(asset_id=AssetKey)
+    def find(self, asset_id):
         """
-        Given a asset list that's a SortedListWithKey, find the index of a particular asset.
+        Find the index of a particular asset in the list.
         Returns: Index of asset, if found. None if not found.
         """
+        # Don't attempt to find an asset by filename in a list that's not sorted by filename.
+        if not self.filename_sort:
+            return None
```

Comment:
I'd probably raise an error, here, since calling `find` on a sorted list with the wrong sort key is an incorrect usage of the class. I would also document that condition in the docstring of `find`.